### ISSUE TITLE
Please add Italian

### ISSUE THREAD

**Issue by MoonDragon-MD** [RHVoice/RHVoice#437]
Hi is it possible to add the Italian language? I found this project based on festival, flite and mbrola:
https://www3.pd.istc.cnr.it/piero/FESTIVAL/home/download-FESTIVAL.htm
I think we can rely on those files to have the Italian language
Thanks

**Comment by paolo-caroni** [comment]
I'm interested too, but maybe it's too early.
The doc is in a revision state (https://github.com/RHVoice/RHVoice/issues/145#issuecomment-644899243).
When they have finisced probably would be easy to add new language/voice. Italian language it's not a simple language however.

**Comment by MoonDragon-MD** [comment]
Yes, Italian is complex, but one can rely on already available "rules". 
Thank you for putting it in the discussion. 
I await the next developments!

**Comment by paolo-caroni** [comment]
It would seem that I am not up to date, there is a guide for adding new entries and new languages in the
[wiki](https://github.com/RHVoice/RHVoice/wiki/Creating-a-new-voice-for-RHVoice.).
I apologize for being misleading.

**Comment by paolo-caroni** [comment]
> Hi is it possible to add the Italian language? I found this project based on festival, flite and mbrola: https://www3.pd.istc.cnr.it/piero/FESTIVAL/home/download-FESTIVAL.htm I think we can rely on those files to have the Italian language Thanks

regarding the reuse of datasets, the Italian language is released under the GPLv2 only license. This creates problems of accounting with MAGE (see License on README).
The Italian language could be released separately, or it would be necessary to start from scratch and create a new voice.
If I want I am available to donate my voice under GPLv3 +.
But I am totally ignorant of how to do it, I have read the guide in the wiki, but I am missing some of the basic concepts.
Have you read and understood the guide? are you interested in creating a new voice?

**Comment by MoonDragon-MD** [comment]
Thank you, I still have to look at the guide, as soon as I try I'll let you know !

**Comment by MoonDragon-MD** [comment]
Hi, unfortunately I stopped after the patch at point "Compiling SPTK."  with the following errors:
`In file included from eps.c:55:
psgr.h:79:15: error: redefinition of ‘struct bbmargin’
   79 | extern struct bbmargin {               /*  Bounding Box Margin  */
      |               ^~~~~~~~
psgr.h:46:8: note: originally defined here
   46 | struct bbmargin {               /*  Bounding Box Margin  */
      |        ^~~~~~~~
psgr.h:80:1: error: expected specifier-qualifier-list before ‘extern’
   80 | extern char *orientations[];
      | ^~~~~~
make[2]: *** [Makefile:370: eps.o] Errore 1
make[2]: uscita dalla directory «/home/md/tts/SPTK-3.11/bin/psgr»
make[1]: *** [Makefile:459: install-recursive] Errore 1
make[1]: uscita dalla directory «/home/md/tts/SPTK-3.11/bin»
make: *** [Makefile:433: install-recursive] Errore 1`

**Comment by zstanecic** [comment]
Hi,

Which distro?

**Comment by MoonDragon-MD** [comment]
Ubuntu  20.04.5 x64  Kernel 5.4.0-126-generic gnome 3.36.8 xorg with nvidia 470.141.03

**Comment by rmcpantoja** [comment]
Hi,
We have a test version of Italian just in case someone is interested.

https://github.com/user-attachments/assets/f3176154-6595-406c-bae8-4c7834d05d64


Cheers.

**Comment by MoonDragon-MD** [comment]
> Hi, We have a test version of Italian just in case someone is interested.
> it_beta_test1.mp4
> 
> Cheers.

Sounds like a good start to me, 
I hope it will be published soon :)

**Comment by rmcpantoja** [comment]
> > Hi, We have a test version of Italian just in case someone is interested.
> > it_beta_test1.mp4
> > Cheers.
> 
> Sounds like a good start to me, I hope it will be published soon :)

Hi,
Thanks. As you can hear, the voice sounding is not good. I would like to get a high-quality Italian voice soon.
But, for now, I am using this one for testing purposes and learn more about the language to improve the support, and I am sure I need help from native Italian speakers to grow the language pack (pronunciation rules). Take account that this is an initial version, so expect errors.
If you would like to collaborate, [contact me from e-mail](mailto:angelitomateocedillo@gmail.com)

Cheers.

**Comment by MoonDragon-MD** [comment]
> > > Hi, We have a test version of Italian just in case someone is interested.
> > > it_beta_test1.mp4
> > > Cheers.
> > 
> > 
> > Sounds like a good start to me, I hope it will be published soon :)
> 
> Hi, Thanks. As you can hear, the voice sounding is not good. I would like to get a high-quality Italian voice soon. But, for now, I am using this one for testing purposes and learn more about the language to improve the support, and I am sure I need help from native Italian speakers to grow the language pack (pronunciation rules). Take account that this is an initial version, so expect errors. If you would like to collaborate, [contact me from e-mail](mailto:angelitomateocedillo@gmail.com)
> 
> Cheers.

Hi, I don't know if I am able to help, however yes, I am Italian. 
Paolo I think can help you more  @paolo-caroni 
Can you make a list of the things you need and write it down here? 
That way we can see what we can help you with.
Thank you for your commitment

**Comment by rmcpantoja** [comment]
> Hi, I don't know if I am able to help, however yes, I am Italian. Paolo I think can help you more @paolo-caroni Can you make a list of the things you need and write it down here? That way we can see what we can help you with. Thank you for your commitment

Hi,
- Any help is appreciated, even an extensive testing of the voice as an Italian speaker and collect all feedback, regarding pronunciation, to improve both language rule system and the voice.
- As I point in the previous message, we'll need better voices (audio/transcription) so the experience can be better for us and the Italian users. Many material as possible.
- And optionally, some more help with phonetics or linguistics. There is still a lot to do, and this is just the beginning. It remains to reflect stressing, long VS. Short vowels handling in our text to phoneme system. But we hope to improve that over time and finally get a stable language pack.

Cheers.

**Comment by MoonDragon-MD** [comment]
>     * Any help is appreciated, even an extensive testing of the voice as an Italian speaker and collect all feedback, regarding pronunciation, to improve both language rule system and the voice.
> 
>     * As I point in the previous message, we'll need better voices (audio/transcription) so the experience can be better for us and the Italian users. Many material as possible.
> 
>     * And optionally, some more help with phonetics or linguistics. There is still a lot to do, and this is just the beginning. It remains to reflect stressing, long VS. Short vowels handling in our text to phoneme system. But we hope to improve that over time and finally get a stable language pack.
>

Ciao,
Okay I can test you and describe what I think should be improved.
Some things would need a grammar teacher, however we can start doing something.
See you soon

**Comment by paolo-caroni** [comment]
>     * As I point in the previous message, we'll need better voices (audio/transcription) so the experience can be better for us and the Italian users. Many material as possible.
> 
About that I have cooperate to the common voice project from mozilla for years, actually it have a good dataset in italian language and probably you can use it.

Honestly if don't have much free time for help, also I'm interested in other TTS such as sherpa-onnx that have a speacking pronunciation like real human voice(actually I'm using that on my smartphone).
But I will help when I can if needed.

**Comment by rmcpantoja** [comment]
Greetings,

> Okay I can test it and describe what I think should be improved.

Thank you. That means a lot. It would be better to contact us via email during the progress, so as not to clutter this discussion.

@paolo-caroni , 
It would be great to try the dataset you mentioned, as long as it has decent quality and a good amount of single-speaker material. If so, please let me know how I can access to it.

Cheers.

**Comment by zstanecic** [comment]
Hi, 
I don't recommend css10. 
Mateo, where i can see you work and the source of it... maybe i can try to help you in improving it, especially when it comes to phoneme flags?

**Comment by zstanecic** [comment]
Btw, i know about one goood italian dataset...

**Comment by rmcpantoja** [comment]
Hi.

@paolo-caroni, 
Thanks for your findings. Yes, in common voice is it possible to select a speaker, but it seems that many speakers, individually, do not have many audio material available to select one, but we could try.
Regarding Multilingual LibrySpeech, which contains LibriVox Material, I found [a book](https://librivox.org/la-divina-commedia-linferno-by-dante-alighieri/) that as hours of high-quality recordings published the past year. Can also be a better option.

@zstanecic, 
Thanks for your interest in improving! I'll add you in the repo for it and seeing where we can improve during development.

Cheers.

**Comment by paolo-caroni** [comment]
> I found [a book](https://librivox.org/la-divina-commedia-linferno-by-dante-alighieri/)

@rmcpantoja
In reality seems the same reader have released multiple [books](https://librivox.org/reader/2033?primary_key=2033&search_category=reader&search_page=1&search_form=get_results&search_order=alpha).

**Comment by rmcpantoja** [comment]
Hi,
> @rmcpantoja In reality seems the same reader have released multiple [books](https://librivox.org/reader/2033?primary_key=2033&search_category=reader&search_page=1&search_form=get_results&search_order=alpha).

Yes, I discovered some previous books and unfortunately has reverb or a small environment in his recordings. But we can use the clean audio of him to build the voice in the future.

@zstanecic, 
I send you the message.

Cheers.

**Comment by Pallas13** [comment]
> Hi.
> 
> @paolo-caroni, 
> Thanks for your findings. Yes, in common voice is it possible to select a speaker, but it seems that many speakers, individually, do not have many audio material available to select one, but we could try.
> Regarding Multilingual LibrySpeech, which contains LibriVox Material, I found [a book](https://librivox.org/la-divina-commedia-linferno-by-dante-alighieri/) that as hours of high-quality recordings published the past year. Can also be a better option.
> 
> @zstanecic, 
> Thanks for your interest in improving! I'll add you in the repo for it and seeing where we can improve during development.
> 
> Cheers.

Have CML-TTS
https://www.openslr.org/146/
